### ISSUE TITLE
ec2_asg doesn't reflect load balancer changes to existing asgs

### ISSUE THREAD

**Issue by jbrockett** [ansible/ansible-modules-core#4460]
##### ISSUE TYPE
- Bug Report
##### COMPONENT NAME

ec2_asg
##### ANSIBLE VERSION

```
ansible 2.0.2.0
  config file = ~/code/ansible.cfg
  configured module search path = Default w/o overrides
```
##### CONFIGURATION

```
[defaults]
ansible_managed = Ansible Managed
display_skipped_hosts = False
forks = 50
gathering = explicit
host_key_checking = False
nocows = 1
retry_files_enabled = False
lookup_plugins = ./lookup_plugins
```
##### OS / ENVIRONMENT

From: OS X El Capitan
Managing: AWS
##### SUMMARY

I am using the ec2_asg module to create an auto-scaling group for my application.  When first created, it had one ELB.  I had to add another ELB after it was created, but it would never update the ASG with the second ELB.  If I create a new asg with 2 ELBs, it works as expected.  
##### STEPS TO REPRODUCE

Use the ec2_asg module to create an ASG with 1 ELB like this:

```
- name: create auto scaling group
  ec2_asg:
    wait_for_instances: false
    name: "test-asg"
    desired_capacity: "{{ asg_desired_capacity }}"
    health_check_period: 600
    health_check_type: EC2
    launch_config_name: "test-lc"
    load_balancers: "{{ ext_elb.elb.name }}"
    min_size: "{{ asg_min_size }}"
    max_size: "{{ asg_max_size }}"
    region: "{{ ec2_region }}"
    state: present
    vpc_zone_identifier: "{{ lookup('tier_subnets', 'vpc_id=' + vpc_id + ' tier=dmz1') }}"
    tags:
      - Name: "test-asg"
      - environment: "{{ ec2_env }}"
```

After the ASG is created, attempt to add another ELB:

```
- name: create auto scaling group
  ec2_asg:
    wait_for_instances: false
    name: "test-asg"
    desired_capacity: "{{ asg_desired_capacity }}"
    health_check_period: 600
    health_check_type: EC2
    launch_config_name: "test-lc"
    load_balancers: "{{ ext_elb.elb.name }},{{ int_elb.elb.name }}"
    min_size: "{{ asg_min_size }}"
    max_size: "{{ asg_max_size }}"
    region: "{{ ec2_region }}"
    state: present
    vpc_zone_identifier: "{{ lookup('tier_subnets', 'vpc_id=' + vpc_id + ' tier=dmz1') }}"
    tags:
      - Name: "test-asg"
      - environment: "{{ ec2_env }}"
```
##### EXPECTED RESULTS

I would expect the ASG to be updated to use 2 ELBs instead of just one.  I do not get any errors when running this.
##### ACTUAL RESULTS

The play runs without issue, but the ASG is not updated to use 2 ELBs.  If I run the exact same play and create a new ASG, both ELBs are present.  This only appears to be an issue with updating an existing ASG.

**Comment by ansibot** [comment]
@ansible ping, this issue is waiting for your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by elnur** [comment]
Experiencing the same problem.

**Comment by ansibot** [comment]
@ansible, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by elnur** [comment]
Except in my case there were no load balancers in an ASG at all and I decided to add one. But nothing changed.

**Comment by shane-walker** [comment]
I'm seeing this same issue, both scenarios listed above.

I took a look at the module and it seems to be something about this block not being completed properly.

``` python
# handle loadbalancers separately because None != []
load_balancers = module.params.get('load_balancers') or []
if load_balancers and as_group.load_balancers != load_balancers:
    changed = True
    as_group.load_balancers = module.params.get('load_balancers')
```

https://github.com/ansible/ansible-modules-core/blob/622ae9c67e1c784996d3ce5ea7632ddd1808c138/cloud/amazon/ec2_asg.py#L509

**Comment by ansibot** [comment]
@ansible, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by jzafran** [comment]
Looks like this is an issue with boto or the AWS API: https://github.com/boto/boto/issues/3428

**Comment by linusgg** [comment]
I'm seeing this same issue. This seems to be caused by a bug in boto 2 and boto 2 is no longer maintained.
Maybe rebuild this module with boto 3 can fix the issue.

**Comment by ansibot** [comment]
@ansible, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by linusgg** [comment]
How about create an alternative module using boto 3 to fix this issue so that it won't introduce breaking change to the current one.
Any thoughts? @bcoca

**Comment by ansibot** [comment]
@ansible, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
@ansible, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
@ansible, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
This repository has been locked. All new issues and pull requests should be filed in https://github.com/ansible/ansible

Please read through the <a href="http://docs.ansible.com/ansible/dev_guide/repomerge.html#move-issues-and-prs-to-new-repo">repomerge</a> page in the dev guide. The guide contains links to tools which automatically move your issue or pull request to the ansible/ansible repo.

**Comment by sunnytinklabs** [comment]
experiencing the same issue

**Comment by spencersmith** [comment]
I'm having the same issue. Is there an update on this? If the boto3 version works, is there a doc for how to implement it? I'm still fairly new to ansible.

**Comment by elnur** [comment]
Is this ever going to be fixed?

**Comment by willthames** [comment]
It's not going to be fixed here. This is a locked repo. There have been a bunch of changes to ec2_asg including a port to boto3 in the last few months. If this is still an issue it will be worth reraising it on ansible/ansible as a new issue.
